src/promptflow/promptflow/exceptions.py

```diff
@@ -209,3 +219,115 @@ class ValidationException(UserErrorException):
     """Exception raised when validation fails."""
 
     pass
+
+
+class _ErrorInfo:
+    @classmethod
+    def get_error_info(cls, e: Exception):
+        if not isinstance(e, Exception):
+            return None, None, None, None, None
+
+        e = cls.select_exception(e)
+        if cls._is_system_error(e):
+            return (
+                ErrorCategory.SYSTEM_ERROR,
+                cls._error_type(e),
+                cls._error_target(e),
+                cls._error_message(e),
+                cls._error_detail(e),
+            )
+        if cls._is_user_error(e):
+            return (
+                ErrorCategory.USER_ERROR,
+                cls._error_type(e),
+                cls._error_target(e),
+                cls._error_message(e),
+                cls._error_detail(e),
+            )
+
+        return ErrorCategory.UNKNOWN, cls._error_type(e), ErrorTarget.UNKNOWN, "", cls._error_detail(e)
+
+    @classmethod
+    def select_exception(cls, e: Exception):
+        """Select the exception  in e and e.__cause__, and prioritize the Exception defined in the promptflow."""
+
+        if isinstance(e, PromptflowException):
+            return e
+
+        # raise Exception("message") from PromptflowException("message")
+        if e.__cause__ and isinstance(e.__cause__, PromptflowException):
+            return e.__cause__
+
+        return e
+
+    @classmethod
+    def _is_system_error(cls, e: Exception):
+        if isinstance(e, SystemErrorException):
```

Comment:
@wangchao1230 , Is it expected that an exception like this, which does not inherit **SystemErrorException** and **UserErrorException**, will return a classification of **UNKNOW**? Do we need to do special handling for this situation to return **SystemErrorException**?
![image](https://github.com/microsoft/promptflow/assets/23182548/a1d557cb-acd3-4885-8d5a-a4498363f71d)